packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -127,17 +133,36 @@ const prepareXAxis = (
   width,
   format,
   orientation = AXES_ORIENTATION.X.BOTTOM,
+  lowerLimit,
+  upperLimit,
 ) => {
   let d3Axis = d3.axisBottom(scale);
   if (isXAxisOrientationTop(orientation)) {
     d3Axis = d3.axisTop(scale);
   }
-  d3Axis
-    .ticks(
-      Math.max(width / constants.MAX_TICK_VARIANCE, constants.MIN_TICKS),
-    )
-    .tickValues(processTickValues(tickValues))
-    .tickFormat(format);
+  if (!tickValues || tickValues.length === 0) {
+    d3Axis
+      .ticks(
+        Math.max(width / constants.MAX_TICK_VARIANCE, constants.MIN_TICKS),
+      )
```

Comment:
Cane we make this a single line for better readability?
```suggestion
      .ticks(Math.max(width / constants.MAX_TICK_VARIANCE, constants.MIN_TICKS))
```